barretenberg/docs/versioned_docs/version-v0.87.0/bb-cli-reference.md

```diff
@@ -0,0 +1,239 @@
+---
+title: Barretenberg CLI Reference
+description: Reference for the bb command-line interface, including commands, options, and usage examples.
+keywords: [bb, CLI, barretenberg, command line, zero-knowledge proofs, proving, verification]
+sidebar_position: 1000000
+---
+
+# Barretenberg CLI Reference
+
+[Barretenberg](https://github.com/AztecProtocol/barretenberg) is Aztec's cryptography backend. This page covers `bb`, the command-line interface for Barretenberg.
+
+Usage: `bb [OPTIONS] [SUBCOMMAND]`
+
+## Installation
+
+To install `bb`, use `bbup`:
+
+```bash
+curl -L https://raw.githubusercontent.com/AztecProtocol/aztec-packages/master/barretenberg/bbup/install | bash
+bbup
+```
+
+## Global options
+
+These options can be used with any `bb` command:
+
+- `-h, --help` - Print help message and exit.
+- `-v, --verbose, --verbose_logging` - Output all logs to stderr.
+- `-d, --debug_logging` - Output debug logs to stderr. 
+- `-c, --crs_path` - Path to  Common Reference String (CRS) directory. CRS files will be downloaded automatically as needed.
+- `--version` - Prints the version string.
+
+## Subcommands
```

Comment:
What's the expected workflow? In what order would a dev typically run these commands? It seems like usage of Noir is implied. What kind of Noir commands do I need to run, in order to generate the files needed for some of the below commands?